Sourcery/Models/Method.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+
+class AMethod: NSObject {
+    
+    class Parameter: NSObject {
+        //external name
+        var argumentLabel: String
+        //internal name
+        var parameterName: String
+        var typeName: String
+        var type: Type?
+
+        init(typeName: String, argumentLabel: String? = nil, parameterName: String) {
+            self.typeName = typeName
+            self.argumentLabel = argumentLabel ?? parameterName
+            self.parameterName = parameterName
+        }
+    }
+    
+    private(set) var parameters: [Parameter]
+    var shortName: String
+    var fullName: String
+    var returnTypeName: String
+    var returnType: Type?
+    var accessLevel: AccessLevel
+    var isStatic: Bool
+    var isClass: Bool
+    
+    /// Annotations, that were created with // sourcery: annotation1, other = "annotation value", alterantive = 2
+    var annotations: [String: NSObject] = [:]
+
+    /// Underlying parser data, never to be used by anything else
+    // sourcery: skipEquality, skipDescription
+    internal var __parserData: Any?
+    // sourcery: skipEquality, skipDescription
+    internal var argumentLabels: [String]
+
+    init(fullName: String,
+         parameters: [Parameter] = [],
+         returnTypeName: String = "Void",
+         accessLevel: AccessLevel = .internal,
+         isStatic: Bool = false,
+         isClass: Bool = false,
+         annotations: [String: NSObject] = [:]) {
+        
+        self.fullName = fullName
+        self.shortName = fullName.range(of: "(").map({ fullName.substring(to: $0.lowerBound) }) ?? fullName
+        self.parameters = parameters
+        self.returnTypeName = returnTypeName
+        self.accessLevel = accessLevel
+        self.isStatic = isStatic
+        self.isClass = isClass
+        
+        if let argumentLabels = fullName.range(of: "(")
```

Comment:
wasn't thinking where it makes most sense, anything preventing us from just doing this in the place where we are creating this entity? I just think this looks like parsing domain and not initializer, `uniqueType` is for final step of processing so if we can just put it where we parse methods instead of waiting until unique types thats better